src/main/java/org/terasology/logic/health/RegenAuthoritySystem.java

```diff
@@ -186,18 +185,20 @@ public void onRegenAddedWithoutComponent(ActivateRegenEvent event, EntityRef ent
             logger.debug("creating new regen component for entity {}", entity);
             RegenComponent regen = new RegenComponent();
             regen.soonestEndTime = Long.MAX_VALUE;
-            addRegenToScheduler(event, entity, regen, health);
+            addRegenToScheduler(event, regen);
             entity.addComponent(regen);
         }
     }
 
-    private void addRegenToScheduler(ActivateRegenEvent event, EntityRef entity, RegenComponent regen,
-                                     HealthComponent health) {
-        if (event.id.equals(BASE_REGEN)) {
-            // setting endTime to -1 because natural regen happens till entity fully regenerates
-            addRegen(BASE_REGEN, health.regenRate, -1, regen);
-        } else {
-            addRegen(event.id, event.value, time.getGameTimeInMs() + (long) (event.endTime * 1000), regen);
+    private void addRegenToScheduler(ActivateRegenEvent event, RegenComponent regen) {
+        if (event.value != 0) {
+            // handle indefinite regeneration actions
+            final long endTime = event.value < 0 ? -1 : time.getGameTimeInMs() + (long) (event.endTime * 1000);
```

Comment:
@jdrueckert I'm pretty sure this is the culprit:

```suggestion
            final long endTime = event.endTime < 0 ? -1 : time.getGameTimeInMs() + (long) (event.endTime * 1000);
```